### ISSUE TITLE
Require padding depending on line count

### ISSUE THREAD

**Issue by ege24** [eslint/eslint#13171]
**What rule do you want to change?**

padded-blocks

**Does this change cause the rule to produce more or fewer warnings?**

It depends.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option: min-expression-count and max-expression-count.
`/*eslint padded-blocks: ["error", "always", { allowSingleLineBlocks: false, minExpressionCount: 3 }]*/`
`/*eslint padded-blocks: ["error", "never", { allowSingleLineBlocks: true, maxExpressionCount: 2 }]*/`

**Please provide some example code that this change will affect:**

```js

  // not allowed
  function foo() {
    bar();
    bar();
    bar();
  }

  // allowed
  function foo() {

    bar();
    bar();
    bar();

  }

  // allowed
  function foo() {
    bar();
    bar();
  }

  // allowed
  function foo() {
    bar();
  }

  // not allowed
  function foo() {

    bar();

  }

  // allowed
  function foo() { bar(); }

  // not allowed
  function foo() { bar(); bar(); bar(); }
```

**What does the rule currently do for this code?**
Disregards the lines in blocks

**What will the rule do after it's changed?**
Consider if a body is long enough to see it worth padding

**Are you willing to submit a pull request to implement this change?**
Yes, if pointed in the right direction

I know it's not the most common style, bu we have a huge codebase that we want to switch to ESLint. It would be great if this is possible.

**Comment by anikethsaha** [comment]
IMO, I do think this might bring inconsistency to the code style as some blocks will be padded and some won't be padded which is the problem this rule is solving.

**Comment by nzakas** [comment]
As mentioned, this is not a common style, so I don’t think it belongs in score rule. I’d suggest creating a custom rule for this purpose.

**Comment by ege24** [comment]
It's a style I've personally seen many times in JS codebases, including my current employer. I'm not sure how to go about this then, should I just fork the current padded-blocks rule?

**Comment by anikethsaha** [comment]
it's not necessary to fork eslint repo. You can simply [write your own eslint plugin](https://eslint.org/docs/developer-guide/working-with-plugins) and you can even extend the eslint's base rule , refer [this example on how to extend a base rule.](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/camelcase.ts)

**Comment by ege24** [comment]
Thanks a lot for the input, I'll try extending the rule.
